src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,375 @@
+            } else if (task instanceof Event) {
+                inputToWrite += SEPARATOR_WORD_FILE;
+                inputToWrite += ((Event) task).getAt();
+            }
+
+            inputToWrite += "\n";
+
+            fw.write(inputToWrite);
+        }
+        fw.close();
+    }
+
+    private void saveTasksToDiskWithException() {
+        try {
+            saveTasksToDisk();
+        } catch (IOException e) {
+            System.out.println("Failed to save new data to the disk");
+        }
+    }
+
+
+    private void addTodoTaskWithException(String taskInput, boolean isForLoading) {
+        try {
+            addTodoTask(taskInput);
+            saveTasksToDiskWithException();
+            if(!isForLoading) {
+                printAddTaskMessage();
+            }
+        } catch (EmptyDescriptionException e) {
+            printHorizontalLine();
+            System.out.println("     ☹ OOPS!!! The description of a todo cannot be empty.");
+            printHorizontalLine();
+        }
+    }
+
+    private void addTodoTask(String taskInput) throws EmptyDescriptionException {
+        String taskName = taskInput.substring(LENGTH_TODO).trim();
+        if (taskName.length() == 0) {
+            throw new EmptyDescriptionException();
+        }
+        tasks.add(new Todo(taskName));
+    }
+
+    private void addDeadlineTask(String taskInput) throws EmptyDescriptionException, MissingInformationException{
+        String taskDescriptionAndDeadline = taskInput.substring(LENGTH_DEADLINE).trim();
+        if (taskDescriptionAndDeadline.length() == 0) {
+            throw new EmptyDescriptionException();
+        }
+
+        String[] taskDescriptionAndDeadlineArray = taskDescriptionAndDeadline.split(SEPARATOR_BY);
+        if (taskDescriptionAndDeadlineArray.length == 1) {
+            throw new MissingInformationException();
+        }
+
+        String taskDescription = taskDescriptionAndDeadlineArray[0];
+        String deadline = taskDescriptionAndDeadlineArray[1];
+        tasks.add(new Deadline(taskDescription, deadline));
+    }
+
+    private void addDeadlineTaskWithException(String taskInput, boolean isForLoading) {
+        try {
+            addDeadlineTask(taskInput);
+            saveTasksToDiskWithException();
+            if (!isForLoading) {
+                printAddTaskMessage();
+            }
+        } catch (EmptyDescriptionException e) {
+            printHorizontalLine();
+            System.out.println("     ☹ OOPS!!! The description of a deadline cannot be empty.");
+            printHorizontalLine();
+        } catch (MissingInformationException e) {
+            printHorizontalLine();
+            System.out.println("     ☹ OOPS!!! Please follow this deadline format: {task_description} /by {deadline}");
+            printHorizontalLine();
+        }
+    }
+
+    private void addEventTask(String taskInput) throws EmptyDescriptionException, MissingInformationException {
```

Comment:
This line is too long ([>120 chars per line](https://se-education.org/guides/conventions/java/basic.html#layout)). Consider splitting this into two lines.